### PR TITLE
Make ocm.go testable

### PR DESCRIPTION
**Pull request by geowa4**
Refactor and add tests to ocm.go. Functionality should be little changed, but I believe I found the cause of a bug and made the correction in ocm.go.

### CODE REVIEW

File: pkg/utils/ocm.go [hunk 1]
```diff
@@ -213,34 +213,43 @@ func loadOCMConfig() (*Config, error) {
...
+	)
 
 	// If neither of the token ENVS 'OCM_TOKEN', 'OCM_REFRESH_TOKEN' are set, or 'OCM_URL' is not set,
 	// use the configuration file as base.
 	// We don't want to always load this, because the user might only use environment variables.
-	if (tokenEnv == "" && refreshTokenEnv == "") || urlEnv == "" {
-		config, err = loadOCMConfig()
-		if err != nil {
-			return &Config{}, fmt.Errorf("Could not load OCM configuration file")
+	if tokenEnv == "" || refreshTokenEnv == "" || urlEnv == "" {
```
Comment (AlexVulaj) [inline pkg/utils/ocm.go:230]: Does this warrant changing the language of the above comment?

Comment (typeid) [reply pkg/utils/ocm.go:230]: +1 on changing the above documentation to explain this change. 
From my understanding we only need either the refresh token or the token alongside an URL for an ocm connection, which is why we had this logic. Is this understanding wrong?

File: pkg/utils/ocm.go [hunk 2]
```diff
@@ -213,34 +213,43 @@ func loadOCMConfig() (*Config, error) {
...
 	}
 	if urlEnv != "" {
 		config.URL = urlEnv
+	} else {
+		config.URL = defaultConfig.URL
 	}
 	if refreshTokenEnv != "" {
 		config.RefreshToken = refreshTokenEnv
+	} else {
+		config.RefreshToken = defaultConfig.RefreshToken
```
Comment (typeid) [inline pkg/utils/ocm.go:252]: What's the benefit here of writing to a new config object instead of overwriting the same potentially already loaded one with the environment variables (if they exist)? It seems less readable to me.

File: pkg/utils/ocm.go (hunk 1 above)
Comment (geowa4) [reply pkg/utils/ocm.go:230]: When I had OCM_TOKEN set and not OCM_REFRESH_TOKEN, I encountered the error below. I'll comment more privately. 

```
FATA[0002] Cannot retrieve clusters: "Get \"https://api.stage.openshift.com/api/clusters_mgmt/v1/clusters?search=omitted&size=50\": can't get access token: invalid_grant: Invalid refresh token" 
```

File: pkg/utils/ocm.go (hunk 2 above)
Comment (geowa4) [reply pkg/utils/ocm.go:251]: I agree with you that it is a bit annoying. I couldn't immediately think of a way to handle optionally loading the config from the file without changing the contract of `loadOcmConfig` to only fill in unset values. But I see now that `loadOcmConfig` is only used once so I could pass in a reference to the config object. 

I also now see that `loadOcmConfig` does not throw an error if the config file is not present. I'm not sure how I feel about that, but we could use that to our advantage and invert the logic here. Is that what you were thinking? 

I think I'm about even on all three options--leaving it as is in this PR, changing to pass a `*Config` reference to `loadOcmConfig`, and overwriting the result of `loadOcmConfig` if the env vars are present. Let me know which direction you prefer, and I'll make the change. Thanks 🙇🏻

Comment (typeid) [reply pkg/utils/ocm.go:251]: >I agree with you that it is a bit annoying. I couldn't immediately think of a way to handle optionally loading the config from the file without changing the contract of loadOcmConfig to only fill in unset values.
Do we need loadOcmConfig to only fill unset values? The way `getOcmConfiguration` currently works, I don't think we do?

>I also now see that loadOcmConfig does not throw an error if the config file is not present.
That's a bit nasty, but also seems intended. It's no big deal here. If the configuration couldn't be loaded, it's an empty configuration, and we will run into an error when trying to establish the connection. I wouldn't worry about it in this PR.

I would prefer leaving it as is in this specific PR, and revert the above block of code to only overwrite the already initiated `Config` object. 

Basically, what we're currently doing is (correct me if I'm wrong here):

```
var config
var tmpConfig

if should_load_ocm_config_file
   tmpConfig = loadOcmConfigFile()

// Check which one has a value with priority and load it into our final config object
// If other values are set by `loadOcmConfigFile`, we lose them.
if OCM_ENV_1 != "" {
    config.ocm_1 = OCM_ENV_1
else {
   config.ocm_1 = tmpConfig.ocm_1
```
vs before
```
var config

if should_load_ocm_config_file
   // Pre-load into the config object
   config = loadOcmConfigFile()

// Overwrite the same config object, in case the env has a value
if OCM_ENV_1 != "" {
    config.ocm_1 = OCM_ENV_1
```

Comment (geowa4) [reply pkg/utils/ocm.go:251]: Ok, I see you. One minute.

Comment (geowa4) [reply pkg/utils/ocm.go]: Updated; pushed; tests still pass.
